lib/Doctrine/MongoDB/Aggregation/Stage/Project.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\MongoDB\Aggregation\Stage;
+
+/**
+ * Fluent interface for adding a $project stage to an aggregation pipeline.
+ *
+ * @author alcaeus <alcaeus@alcaeus.org>
+ */
+class Project extends Operator
```

Comment:
Absolutely. The ```Operator``` class defines all stage operators that are common to ```$group```, ```$redact``` and ```$project```. This can be used to reshape documents as well, as you can see in the example test: https://github.com/doctrine/mongodb/pull/213/files#diff-41873c0811e1fca2085b746a594da896R107.
As for ```addAnd()``` and ```addOr()```, they absolutely make sense in that class and can be used in $project stages as well:
```
{
    $project: {
        someField: { $and: [ '$field', '$otherField' ] }
    }
}
```

However, due to a bug in my brain this is not possible since the ```$and``` and ```$or``` operators don't respect the current field - I'll fix this along with the other issues.